modules/acquia_cms_common/src/EventSubscriber/KernelTerminate/AcquiaCmsTelemetry.php

```diff
@@ -251,16 +278,86 @@ private function shouldSendTelemetryData(): bool {
     if ($isCI) {
       return FALSE;
     }
-    if (AcquiaDrupalEnvironmentDetector::isAhEnv()) {
-      $telemetryOpted = $this->state->get('acquia_connector.telemetry.opted', TRUE);
-      $isAcquiaTelemetryDataSent = $this->state->get('acquia_cms_telemetry.status', FALSE);
-      if ($telemetryOpted && !$isAcquiaTelemetryDataSent) {
-        return TRUE;
-      }
+    $sendTimestamp = $this->state->get('acquia_cms_common.telemetry.timestamp');
+    if (AcquiaDrupalEnvironmentDetector::isAhEnv() &&
+    $this->state->get('acquia_connector.telemetry.opted', TRUE) &&
+    $this->time->getCurrentTime() - $sendTimestamp > 86400 &&
+    $this->shouldResendData()) {
+      return TRUE;
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * Decides if telemetry data should resend or not.
+   */
+  private function shouldResendData(): bool {
+    $telemetryDataOld = $this->state->get('acquia_cms_common.telemetry.data', []);
+    if (!empty($telemetryDataOld)) {
+      $telemetryDataOld = json_decode($telemetryDataOld, TRUE);
+    }
+    $telemetryDataCurrent = $this->getAcquiaCmsTelemetryData();
+    if (empty($telemetryDataOld) || (is_array($telemetryDataOld) &&
+    $this->isTelemetryDataDifferent($telemetryDataOld, $telemetryDataCurrent))) {
+      return TRUE;
     }
     return FALSE;
   }
 
+  /**
+   * Check difference between old and new telemetry data.
+   */
+  private function isTelemetryDataDifferent($array1, $array2) {
+    $difference = [];
+    foreach ($array1 as $key => $value) {
+      if (is_array($value)) {
+        if (!isset($array2[$key]) || !is_array($array2[$key])) {
+          $difference[$key] = $value;
+        }
+        else {
+          $new_diff = $this->isTelemetryDataDifferent($value, $array2[$key]);
+          if (!empty($new_diff)) {
+            $difference[$key] = $new_diff;
+          }
+        }
+      }
+      elseif (!array_key_exists($key, $array2) || $array2[$key] !== $value) {
+        $difference[$key] = $value;
+      }
+    }
+    return $difference;
+  }
+
+  /**
+   * Get an array of information about Lightning extensions.
+   *
+   * @return array
+   *   An array of extension info keyed by the extensions machine name. E.g.,
+   *   ['lightning_layout' => ['version' => '8.2.0', 'status' => 'enabled']].
+   */
+  private function getExtensionInfo() {
```

Comment:
@rajeshreeputra There should be PHPUnit tests for this function as well.

Reply:
could you please help me here for getExtensionInfo() method what to check in test? 
- data returned by this method is not empty? OR
```$method = $this->getAcqauiaCmsTelemetryMethod("getExtensionInfo");
  // Remove `CI` environment variable, or we can set it to false.
  putenv("CI=");
  $state_service->set("acquia_connector.telemetry.opted", TRUE);
  $extensions = $method->invoke($this->acquiaCmsTelemetry);
  
  $this->assertNotEmpty($extensions, "Extensions available.");
```
- data returned by this method is array? OR
```$method = $this->getAcqauiaCmsTelemetryMethod("getExtensionInfo");
  // Remove `CI` environment variable, or we can set it to false.
  putenv("CI=");
  $state_service->set("acquia_connector.telemetry.opted", TRUE);
  $extensions = $method->invoke($this->acquiaCmsTelemetry);
  
  $this->assertIsArray($extensions, "Extensions available."); assertIsArray
```
- compare list of all module with data returned by this method like:
``` $acquia_modules = [
  'acquia_cms_audio' => [
    'status' => 'disabled',
    'version' => 'dev',
  ],
  .
  .
  .
  'acquia_connector' => [
    'status' => 'enabled',
    'version' => '4.0.4',
  ],
 'cohesion' => [
    'status' => 'disabled',
    'version' => '7.2.2',
  ],
  .
  .
  .
  'sitestudio_config_management' => [
    'status' => 'disabled',
    'version' => 'dev',
  ]
  $method = $this->getAcqauiaCmsTelemetryMethod("getExtensionInfo");
  // Remove `CI` environment variable, or we can set it to false.
  putenv("CI=");
  $state_service->set("acquia_connector.telemetry.opted", TRUE);
  $extensions = $method->invoke($this->acquiaCmsTelemetry);
  $this->assertTrue($acquia_modules === $extensions, "Extensions matches.");
  ```

Reply:
done.